src/openeo_gfmap/features/feature_extractor.py

```diff
@@ -0,0 +1,285 @@
+"""Feature extractor functionalities. Such as a base class to assist the
+implementation of feature extractors of a UDF.
+"""
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+import openeo
+import xarray as xr
+from openeo.udf import XarrayDataCube
+from openeo.udf.run_code import execute_local_udf
+from openeo.udf.udf_data import UdfData
+
+REQUIRED_IMPORTS = """
+import inspect
+from abc import ABC, abstractmethod
```

Comment:
why are inspect and abc required imports?

Reply:
inspect is useless in this case, I removed it, but ABC is used as it defines the base class for all feature extractors

Reply:
I'm actually wondering if you can't eliminate the whole `REQUIRED_IMPORTS` by replacing those `inspect.getsource(...)` constructs with just concatenating the python files that define these classes, then all your dependencies are handled automatically and you don't have to manually keep  a `REQUIRED_IMPORTS` in sync

Reply:
Moreover, I think it's useful to have unit tests on that UDF source code building  pipeline, e.g. check that the generated UDF code can at least be parsed (e.g. to just parse with executing with `ast.parse` from `ast` module or full compile with `compile()` built-in), and ideally also check that it can be loaded (e.g. all imports are valid, ...)